### ISSUE TITLE
Configure issue with MVAPICH2 on Omni-Path

### ISSUE THREAD

**Issue by peterinl** [easybuilders/easybuild-easyconfigs#7167]
I am having trouble configuring/building MVAPICH2-2.3 on our new omnipath system. I add the configopt "--with-device=ch3:psm", but it seems easybuild is adding "--with-rdma=gen2" at some time during the configure. Apparently those two options are mutually exclusive, and the make fails. I tried adding "--without-rdma" but it gets overridden. Perhaps this is user error, but I can't figure out how to do away with the "--with-rdma=gen2" option.

I am on a CentOS 7.5, LMOD 7.8.
[easybuild.log](https://github.com/easybuilders/easybuild-easyconfigs/files/2585898/easybuild.log)

command used: "eb MVAPICH2-2.3-GCC-7.3.0-2.30.eb" (see contents below)

I am using EasyBuild 3.7.1
```
buildpath          (E) = /apps/local/easybuild/build
containerpath      (E) = /apps/local/easybuild/containers
debug              (E) = True
include-easyblocks (E) = /hpc-common/software/easybuild/inl-easybuild/easybuild/easyblocks/*.py
installpath        (E) = /apps/local/easybuild
minimal-toolchains (E) = True
packagepath        (E) = /apps/local/easybuild/packages
prefix             (E) = /apps/local/easybuild
repositorypath     (E) = /apps/local/easybuild/ebfiles_repo
robot-paths        (E) = /hpc-common/software/easybuild/inl-easybuild/easybuild/easyconfigs, /hpc-common/software/EasyBuild/3.7.1/lib/python2.7/site-packages/easybuild_easyconfigs-3.7.1-py2.7.egg/easybuild/easyconfigs
sourcepath         (E) = /hpc-common/software/easybuild/sources
```

Using this easyconfig:
```
name = 'MVAPICH2'
version = '2.3'

homepage = 'http://mvapich.cse.ohio-state.edu/overview/mvapich2/'
description = "This is an MPI 3.0 implementation.  It is based on MPICH2 and MVICH."

toolchain = {'name': 'GCC', 'version': '7.3.0-2.30'}

source_urls = ['http://mvapich.cse.ohio-state.edu/download/mvapich/mv2/']
sources = [SOURCELOWER_TAR_GZ]

# Let's store the checksum in order to be sure it doesn't suddenly change
#checksums = ['5651e8b7a72d7c77ca68da48f3a5d108']

builddependencies = [('Bison', '3.0.4')]

configopts = '--with-device=ch3:psm --disable-mcast'

moduleclass = 'mpi'
```

**Comment by boegel** [comment]
@peterinl Thanks for reporting this!

The problem is basically that `--with-rdma` is hardcoded in the `MVAPICH2` easyblock currently. You can control the value that is passed to it via the `rdma_type` easyconfig parameter (`gen2` is the default value).

With the fairly trivial change I made to the `MVAPICH2` easyblock in https://github.com/easybuilders/easybuild-easyblocks/pull/1586, you can use `rdma_type = None` to make it use `--without-rdma` instead.

If you want to test this with EasyBuild v3.7.1:

1) Download the updated easyblock:
```
curl -OL https://raw.githubusercontent.com/easybuilders/easybuild-easyblocks/22947b91e29fb1521b52772b9090fdca6b138a3b/easybuild/easyblocks/m/mvapich2.py
```

2) Pass it to `eb` via `--include-easyblocks`:
```bash
eb MVAPICH2-2.3-GCC-7.3.0-2.30.eb --include-easyblocks mvapich2.py
```

Well get the updated `MVAPICH2` easyblock included with the upcoming EasyBuild v3.8.0.

**Comment by boegel** [comment]
Fixed with https://github.com/easybuilders/easybuild-easyblocks/pull/1586 (which will be included in EasyBuild v3.8.0). You can bypass the default use `--with-rdma` via `rdma_type = None` in the easyconfig file.
